test/testdata/infer/unreachable_code_instance_variable.rb

```diff
@@ -0,0 +1,13 @@
+# typed: true
+
+extend T::Sig
+
+class C
+  def initialize
+    @x = T.let(42, Integer)
+  end
+
+  def f
+    @x ||= 43 # error: This code is unreachable
+  end
```

Comment:
I think this case is _not_ covered in `ivar_cvar_oror_eq.rb` (`g1` and `g2` only deal with nilable case) but it might make sense to fold it into that file.